packages/app/src/systems/Account/components/BalanceWidget/BalanceWidget.tsx

```diff
@@ -0,0 +1,152 @@
+import { cssObj } from "@fuel-ui/css";
+import { Avatar, Box, Flex, Icon, IconButton, Text } from "@fuel-ui/react";
+import type { ReactNode } from "react";
+
+import type { Account } from "../../types";
+
+import { BalanceWidgetLoader } from "./BalanceWidgetLoader";
+
+import { shortAddress } from "~/systems/Core";
+
+type BalanceWidgetWrapperProps = {
+  children: ReactNode;
+};
+
+export function BalanceWidgetWrapper({ children }: BalanceWidgetWrapperProps) {
+  return (
+    <Box css={styles.balanceWidgetWrapper}>
+      <Box css={styles.backgroundShadow}>&nbsp;</Box>
+      <Box css={styles.backgroundFront}>&nbsp;</Box>
+      <Flex css={styles.contentWrapper}>{children}</Flex>
+    </Box>
+  );
+}
+
+export type BalanceWidgetProps = {
+  account: Account;
+  isHidden?: boolean;
+};
+
+export function BalanceWidget({ account, isHidden }: BalanceWidgetProps) {
+  async function handleCopy() {
+    await navigator.clipboard.writeText(account.address);
+  }
+
+  return (
+    <BalanceWidgetWrapper>
+      <Flex direction="column" align="center">
+        <Avatar
+          size="lg"
+          name={account.name}
+          /**
+           * TODO: Need to add Avatar.Generated here when it's done on @fuel-ui
+           */
+          src="https://images.unsplash.com/photo-1492633423870-43d1cd2775eb?&w=128&h=128&dpr=2&q=80"
+        />
+        <IconButton
+          size="xs"
+          variant="ghost"
+          color="gray"
+          icon={<Icon icon="CaretDown" color="gray8" />}
+          aria-label="Expand"
+          css={styles.caretDownIcon}
+        />
+      </Flex>
+      <Flex justify="space-between" css={{ flex: "1 0" }}>
+        <Flex direction="column" css={{ marginLeft: 14, alignSelf: "center" }}>
+          <Flex align="center">
+            <Text fontSize="sm" color="gray11" css={{ fontWeight: "bold" }}>
+              {shortAddress(account.address)}
+            </Text>
+            <IconButton
```

Comment:
We should use the copy component here: https://fuellabs.github.io/fuel-ui/?path=/story/helpers-copyable--usage